lua/dap/ext/autocompl.lua

```diff
@@ -0,0 +1,57 @@
+local M = {}
+local api = vim.api
+local dap = require('dap')
+local timer = nil
+
+
+local function destroy_timer()
+  if timer then
+    timer:stop()
+    timer:close()
+    timer = nil
+  end
+end
+
+
+local function trigger_completion()
+  dap.omnifunc(1, "")
+end
+
+
+function M._InsertCharPre()
+  destroy_timer()
+  local char = api.nvim_get_vvar('char')
+  if tonumber(vim.fn.pumvisible()) == 1 then
+    return
+  end
+  local session = dap.session()
+  if not session then
+    return
+  end
+  local triggers = session.capabilities.completionTriggerCharacters or {'.'}
```

Comment:
Maybe add here a check for `sessions.capabilities` being non-nil. I'm getting error when the session failed to start and therefore has no capabilities.

Reply:
Thank you for pointing out :+1: 

Adapted it to handle the nil case.